src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileStatusService.scala

```diff
@@ -1,14 +1,40 @@
 package uk.gov.nationalarchives.tdr.api.service
 
+import uk.gov.nationalarchives.Tables.FilestatusRow
 import uk.gov.nationalarchives.tdr.api.db.repository.{DisallowedPuidsRepository, FileStatusRepository}
-import uk.gov.nationalarchives.tdr.api.service.FileStatusService.{Antivirus, ChecksumMatch, FFID, Success}
+import uk.gov.nationalarchives.tdr.api.graphql.DataExceptions.InputDataException
+import uk.gov.nationalarchives.tdr.api.graphql.fields.FileStatusFields.{AddFileStatusInput, FileStatus}
+import uk.gov.nationalarchives.tdr.api.service.FileStatusService.{Antivirus, ChecksumMatch, FFID, Failed, Success, Upload}
 
+import java.sql.Timestamp
+import java.time.Instant
 import java.util.UUID
 import scala.concurrent.{ExecutionContext, Future}
 
 class FileStatusService(
                          fileStatusRepository: FileStatusRepository,
-                         disallowedPuidsRepository: DisallowedPuidsRepository)(implicit val executionContext: ExecutionContext) {
+                         disallowedPuidsRepository: DisallowedPuidsRepository,
+                         uuidSource: UUIDSource)(implicit val executionContext: ExecutionContext) {
+
+  def addFileStatus(addFileStatusInput: AddFileStatusInput): Future[FileStatus] = {
+
+    validateStatusTypeAndValue(addFileStatusInput)
+    for {
+      _ <- fileStatusRepository.addFileStatuses(List(FilestatusRow(uuidSource.uuid, addFileStatusInput.fileId, addFileStatusInput.statusType,
+        addFileStatusInput.statusValue, Timestamp.from(Instant.now()))))
+    } yield FileStatus(addFileStatusInput.fileId, addFileStatusInput.statusType, addFileStatusInput.statusValue)
+  }
+
+  private def validateStatusTypeAndValue(addFileStatusInput: AddFileStatusInput) = {
+    val statusType: String = addFileStatusInput.statusType
+    val statusValue: String = addFileStatusInput.statusValue
+
+    if (Upload == statusType && List(Success, Failed).contains(statusValue)) {
+      true
+    } else {
+      throw InputDataException(s"Invalid FileStatus input: either '$statusType' or '$statusValue'")
```

Comment:
This is personal preference so feel free to ignore me. Throwing exceptions isn't very functional (I know there's a lot thrown in here already)
If you return `Future[Boolean]` from this method then you can do 
```scala
    if (Upload == statusType && List(Success, Failed).contains(statusValue)) {
      Future(true)
    } else {
      Future.failed(InputDataException(s"Invalid FileStatus input: either '$statusType' or '$statusValue'"))
    }
```

That will let you use it inside your for comprehension as well
```scala
    for {
      _ <- validateStatusTypeAndValue(addFileStatusInput)
      _ <- fileStatusRepository.addFileStatuses(List(FilestatusRow(uuidSource.uuid,...
```

Reply:
Yeah, make sense. Thanks.